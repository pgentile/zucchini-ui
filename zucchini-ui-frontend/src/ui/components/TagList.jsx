import PropTypes from "prop-types";
import React from "react";

import ListWithSeparator from "./ListWithSeparator";
import Tag from "./Tag";

export default class TagList extends React.PureComponent {
  static propTypes = {
    testRunId: PropTypes.string.isRequired,
    tags: PropTypes.arrayOf(PropTypes.string).isRequired
  };

  render() {
    const { tags, testRunId } = this.props;

    // Sort tag list before display
    const sortedTags = [...tags];
    sortedTags.sort();

    const tagElements = sortedTags.map((tag) => <Tag key={tag} testRunId={testRunId} tag={tag} />);

    return <ListWithSeparator separator=" ">{tagElements}</ListWithSeparator>;
  }
}
