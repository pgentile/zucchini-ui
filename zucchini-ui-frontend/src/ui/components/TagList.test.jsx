import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import TagList from "./TagList";

describe("TagList", () => {
  it("should contain labels with tags", () => {
    const tags = ["A", "B", "C"];
    const testRunId = "sampleId";

    const { asFragment } = render(
      <MemoryRouter>
        <TagList testRunId={testRunId} tags={tags} />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
