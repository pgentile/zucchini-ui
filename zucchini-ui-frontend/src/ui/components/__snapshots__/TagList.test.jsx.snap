// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagList should contain labels with tags 1`] = `
[
  <a
    className="badge badge-info"
    href="/test-runs/sampleId/tag-details?tag=A"
    onClick={[Function]}
  >
    @
    A
  </a>,
  " ",
  <a
    className="badge badge-info"
    href="/test-runs/sampleId/tag-details?tag=B"
    onClick={[Function]}
  >
    @
    B
  </a>,
  " ",
  <a
    className="badge badge-info"
    href="/test-runs/sampleId/tag-details?tag=C"
    onClick={[Function]}
  >
    @
    C
  </a>,
]
`;
